lib/shared/utils.js

```diff
@@ -48,22 +48,15 @@ function timecodeToMicroseconds(timecode, fps) {
 }
 
 function extractStyling(text) {
-  let unformattedText = '';
-  if (text.search(/<br>/) && !text.startsWith('<br>')) {
-    unformattedText = text.replace(/<br>/, '\n');
-  }
-  const formattingTags = unformattedText.match(/<[^>]*>|{[^}]*}/g);
-  if (formattingTags && formattingTags.length) {
-    formattingTags.forEach(tag => {
-      if (tag.search(/{[^}]*}/) > -1) {
-        unformattedText = unformattedText.replace(tag, ' ').replace(/\s\s+/, ' ');
-      } else {
-        unformattedText = unformattedText.replace(tag, '');
-      }
-    });
-  }
-
-  return unformattedText.trim();
+  const regexReplace = [
+    [/(?<!^)<br>/g, '\n'],
+    [/<.*?>/g, ''],
+    [/{.*?}/g, ' '],
+    [/((?<!--)>|<|{|})/g, ''],
+    [/ {2,}/g, ' '],
+    [/^\s+|\s+$/gm, ''],
+  ];
+  return regexReplace.reduce((newText, re) => newText.replace(re[0], re[1]), text);
```

Comment:
Great use of reduce!

Reply:
This is not too much of an issue here, but at a larger scale it could be cumbersome to track down what `re[0]` and `re[1]` are.  It is a good practice to give values a relevant name so the code is more readable.  In some cases I might just take the extra step of creating a variable i.e. `const regex = re[0]; const value = re[1]`.

In this case it might make sense to change `replaceText` into an array of objects i.e. 
```const replaceText = [
  { regex: /<br>/, value: ''},
  ...
];
```
This would allow you to adjust this line into something more readable.  i.e.
`return regexReplace.reduce((newText, {regex, value}) => newText.replace(regex, value), text);`